app/routes.py

```diff
@@ -1,29 +1,95 @@
+from app import app, db
+from app.calculations import (
+    get_amount_negative,
+    get_amount_positive,
+    get_negative_percent,
+    get_positive_percent,
+)
 from app.forms import MovieSearchForm
 from app.models import MovieInfo
-from flask import redirect, render_template, url_for
+from app.pdf_creator import create_history_pdf
+from app.predictions import get_predictions
+from app.scraper import get_all_movies
+from flask import redirect, render_template, request, url_for
+from sqlalchemy.exc import SQLAlchemyError
 
 
 @app.route("/", methods=["GET", "POST"])
 @app.route("/main", methods=["GET", "POST"])
 def main_page() -> str:
     """Get a main page with movie search."""
     form = MovieSearchForm()
-    if form.validate_on_submit():
-        return redirect(url_for("result"))
+    if request.method == "POST":
+        title = form.movie_title.data
+        movies = get_all_movies(title)
+        amount = len(movies)
+        if amount == 0:
+            return redirect(url_for("error_handler", erorr="No movie found."))
+        elif amount == 1:
+            return redirect(url_for("result", movie=movies[0]))
+        else:
+            return redirect(url_for("search_results", movies=movies))
     return render_template("main.html", form=form)
 
 
-@app.route("/result", methods=["GET", "POST"])
-def result() -> str:
+@app.route("/result/<tuple:movie>", methods=["GET", "POST"])
+def result(movie: Tuple[str, str]) -> str:  # noqa: CCR001 8 > 5
     """Get page with review analysis information."""
-    movie = {"name": "Game of Thrones", "pos_per": 80, "neg_per": 20}
-    return render_template("result.html", movie=movie)
+    correct_title, movie_id = movie
+    if db.session.query(db.exists().where(MovieInfo.title == correct_title)).scalar():
+        movie = (
+            MovieInfo.query.order_by(MovieInfo.movie_id.desc())
+            .filter_by(title=correct_title)
+            .first()
+        )
+        if (datetime.today() - movie.search_date).days < 30:
+            new_movie = MovieInfo(
+                title=correct_title,
+                positive_percent=movie.positive_percent,
+                negative_percent=movie.negative_percent,
+                amount_positive_reviews=movie.amount_positive_reviews,
+                amount_negative_reviews=movie.amount_negative_reviews,
+            )
+    else:
+        try:
+            predictions = get_predictions(movie_id)
+        except Exception:
+            return redirect(url_for("error_handler", error="No reviews yet."))
+        new_movie = MovieInfo(
+            title=correct_title,
+            positive_percent=get_positive_percent(predictions),
+            negative_percent=get_negative_percent(predictions),
+            amount_positive_reviews=get_amount_positive(predictions),
+            amount_negative_reviews=get_amount_negative(predictions),
+        )
+    try:
+        db.session.add(new_movie)
+        db.session.commit()
+    except SQLAlchemyError:
```

Comment:
You can add error handlers to flask so you will not need to handle SQLAlchemyError error manually. (Optional)